### ISSUE TITLE
Specify the current directory of the debugger

### ISSUE THREAD

**Issue by actboy168** [Microsoft/vscode#54465]
I am using a process to implement the debugger, like this
``` json
{
    "program": ".\\bin\\lua.exe",
    "args": [
        ".\\script\\main.lua"
    ]
}
```

For `program`, its relative path is based on the extension directory. But for `args`, its relative path is based on the vscode directory. This directory has no value for the extension, so I want to specify the current directory of the program.

**Comment by weinand** [comment]
@actboy168 what does your JSON snippet from above show?
A part of the extension's package.json or a launch config from the launch.json?

**Comment by actboy168** [comment]
package.json

**Comment by weinand** [comment]
why are you trying to "reach" outside of your extension folder for one argument (the `.\\script\\main.lua`)?

**Comment by actboy168** [comment]
".\\script\\main.lua" is the code for my debugger, not the code to debug target.

**Comment by weinand** [comment]
If I remember correctly then a debug adapter executable is specified like this:
```
	"runtime": ".\\bin\\lua.exe",
	"program": ".\\script\\main.lua"
```

**Comment by actboy168** [comment]
According to the docs, runtime only supports two options: `node` and `mono`.

https://github.com/Microsoft/vscode-docs/blob/master/docs/extensions/example-debuggers.md

**Comment by actboy168** [comment]
I think my `package.json` is fine. VSCode does create a `lua.exe` process and correctly applies `args` as a command line argument. The problem is that the current directory of this process is the directory of VSCode, and I want to change it.

**Comment by weinand** [comment]
The doc might be outdated. The implementation probably allows any runtime.
Please try it.

**Comment by actboy168** [comment]
OK, `runtime` is also work. But the current directory of the process is still the directory of VSCode.`program` will be rewritten as an absolute path based on the extension directory, which will solve some of my problems. I still need to know where the extension directory is.

**Comment by weinand** [comment]
This is fixed by the new API made available as part of #56646
